jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/assignment/impl/strategy/RoundRobinAssignmentStrategy.java

```diff
@@ -0,0 +1,118 @@
+                super.offer(headValue);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            return headValue;
+        }
+    }
+
+    @Override
+    public String getIdentifier() {
+        return IDENTIFIER;
+    }
+
+    @Override
+    public Assignment apply(Task task, TaskContext taskContext, String s) {
+        List<OrganizationalEntity> excluded = ((InternalPeopleAssignments)task.getPeopleAssignments()).getExcludedOwners();
+        UserInfo userInfo = (UserInfo) ((org.jbpm.services.task.commands.TaskContext)taskContext).get(EnvironmentName.TASK_USER_INFO);
+
+        // Get the the users from the task's the potential owners
+        List<OrganizationalEntity> potentialOwners = task.getPeopleAssignments().getPotentialOwners().parallelStream()
+                .filter(oe -> oe instanceof User && !excluded.contains(oe))
+                .collect(Collectors.toList());
+
+        // Get the users belonging to groups that are potential owners
+        task.getPeopleAssignments().getPotentialOwners().parallelStream().filter(oe -> oe instanceof Group)
+                .forEach(oe -> {
+                    Iterator<OrganizationalEntity> groupUsers = userInfo.getMembersForGroup((Group)oe);
+                    if (groupUsers != null) {
+                        groupUsers.forEachRemaining(user -> {
+                            if (user != null && !excluded.contains(user) && !potentialOwners.contains(user)) {
+                                potentialOwners.add(user);
+                            }
+                        });
+                    }
+                });
+        String queueName = getQueueName(task);
+        CircularQueue<OrganizationalEntity> mappedQueue = null;
+        // If a queue already exists for this task then its contents should be synchronized with the
+        // current list of potential owners
+        if (circularQueueMap.containsKey(queueName)) {
+            final CircularQueue<OrganizationalEntity> queue = circularQueueMap.get(queueName); 
+            potentialOwners.forEach(po -> {
+            	if (!queueContainsUser(queue,po)) {
+            		queue.add(po);
+            	}
+            });
+            queue.removeIf(oe -> !potentialOwners.contains(oe));
+            mappedQueue = queue;
+        } else {
+            CircularQueue<OrganizationalEntity> queue = new CircularQueue();
+            potentialOwners.forEach(po -> {queue.add(po);});
+            circularQueueMap.put(queueName,queue);
+            mappedQueue = queue;
+        }
+        OrganizationalEntity owner = mappedQueue.take();
+        return new Assignment(owner.getId());
+    }
+    
+    protected boolean queueContainsUser(CircularQueue<OrganizationalEntity> queue, OrganizationalEntity oe) {
+    	return queue.contains(oe);
+    }
+
+    protected String getQueueName(Task task) {
+        return task.getTaskData().getDeploymentId()+"_"+task.getName();
```

Comment:
I'd recommend to add process id to the name as well as there might be tasks with same name in different processes but still in the same kjar